cpp-terminal/base.hpp

```diff
@@ -1,109 +1,238 @@
 #pragma once
 
+#include <cstdint>
 #include <string>
-// TODO: remove include
-#include "private/platform.hpp"
-
+#include "cpp-terminal/private/platform.hpp"
 namespace Term {
-
-enum class style {
-    reset = 0,
-    bold = 1,
-    dim = 2,
-    italic = 3,
-    underline = 4,
-    blink = 5,
-    blink_rapid = 6,
-    reversed = 7,
-    conceal = 8,
-    crossed = 9,
-    overline = 53
+/*
+ * The 3bit/4bit colors for the terminal
+ * get the foreground color: Color4 + 30, Background color: Color4 + 40
+ * See https://en.wikipedia.org/wiki/ANSI_escape_code#3-bit_and_4-bit
+ */
+enum class Color4 : uint8_t {
+    // FG: 30, BG: 40
+    BLACK = 0,
+    // FG: 31, BG: 41
+    RED = 1,
+    // FG: 32, BG: 42
+    GREEN = 2,
+    // FG: 33, BG: 43
+    YELLOW = 3,
+    // FG: 34, BG: 44
+    BLUE = 4,
+    // FG: 35, BG: 45
+    MAGENTA = 5,
+    // FG: 36, BG: 46
+    CYAN = 6,
+    // FG: 37, BG: 47
+    WHITE = 7,
+    // disables color, FG: 39, BG: 49
+    NONE = 9,
```

Comment:
Maybe call it DEFAULT ?

Reply:
Implemented in b422dd2b80a2c7d7b170f1f5743959ca9245e5fa.